trafaret/base.py

```diff
@@ -1246,36 +1270,39 @@ def __repr__(self):
         return "<Mapping(%r => %r)>" % (self.key, self.value)
 
 
-class Enum(Trafaret):
+class Enum(Literal):
     """
-    >>> trafaret = Enum("foo", "bar", 1) >> ignore
-    >>> trafaret
-    <Enum('foo', 'bar', 1)>
-    >>> trafaret.check("foo")
-    >>> trafaret.check(1)
-    >>> extract_error(trafaret, 2)
-    "value doesn't match any variant"
-    >>> trafaret.is_valid(1)
-    True
-    >>> trafaret.is_valid(2)
-    False
+    Deprecated: use Literal.
     """
-    __slots__ = ['variants']
+    def __init__(self, *args, **kwargs):
+        deprecated("Use Literal.")
```

Comment:
Can you explain this? Maybe my english skill are not good enough to understand why we should deprecate `Enum` in favour of `Literal`.

Reply:
As mentioned above, I thought it would be useful to be consistent with the typing module (most of the rest of Trafaret looks quite similar): https://docs.python.org/3/library/typing.html

Also, Enum is not a great name, as it's nothing to do with `enum.Enum`, and could cause confusion with the new `ToEnum`.